### PR TITLE
Coco segmentation

### PR DESCRIPTION
**Pull request by Louis-Dupont**

### CODE REVIEW

File: src/data_gradients/datasets/__init__.py [hunk 1]
```diff
@@ -8,5 +13,7 @@
     "YoloFormatDetectionDataset",
     "VOCSegmentationDataset",
     "VOCFormatSegmentationDataset",
+    "CocoSegmentationDataset",
```
Comment (BloodAxe) [inline src/data_gradients/datasets/__init__.py:16]: Coco or COCO?

File: tests/unit_tests/datasets/segmentation/test_coco_segmentation_dataset.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import unittest
+from torch.utils.data import DataLoader
+from data_gradients.managers.segmentation_manager import SegmentationAnalysisManager
+from data_gradients.datasets.segmentation.coco_segmentation_dataset import CocoSegmentationDataset
+
+
+class CocoSegmentationDatasetTest(unittest.TestCase):
+    def setUp(self):
+        self.train_set = CocoSegmentationDataset(root_dir="../../../../example_dataset/tinycoco", split="train", year="2017")
```
Comment (BloodAxe) [inline tests/unit_tests/datasets/segmentation/test_coco_segmentation_dataset.py:9]: Let's figure out path using `os.path.dirname(__file__)` and not assume that tests will be run from the same dir. 
If I run `python -m unit_test tests.unit_tests.datasets....` that would crash since cwd would be different.

Comment (Louis-Dupont) [reply tests/unit_tests/datasets/segmentation/test_coco_segmentation_dataset.py]: Changed in 92519f5f088612d83137a58c7b98e607c1075d9f

File: src/data_gradients/datasets/__init__.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/data_gradients/datasets/__init__.py]: Changed in [4f0c550](https://github.com/Deci-AI/data-gradients/pull/130/commits/4f0c5503f7627f5bb1da8b171675e19ced5c588b)

**Review by BloodAxe — APPROVED** [review]
LGTM
